twtr/src/integration-test/groovy/twtr/FollowFunctionalTests/FollowHappySpec.groovy

```diff
@@ -0,0 +1,275 @@
+                       emailAddress: goodEmailAccount + postfix + goodEmailDomain,
+                       password    : goodPassword,
+                       displayName : goodDisplayName + postfix])
+        return response.data.id
+    }
+
+    def 'only two are available with no followers or following'() {
+        when:
+        goodId1 = addAccount('1')
+        goodId2 = addAccount('2')
+        def response = restClient.get(path: '/accounts')
+
+        then:
+        response.status == 200
+        response.data.size() == 2
+
+        when:
+        response = restClient.get(path:"/accounts/${goodId1}")
+
+        then:
+        response.status == 200
+        response.data
+        response.data.followerCount == 0
+        response.data.followingCount == 0
+        response.data.id == goodId1
+
+        when:
+        response = restClient.get(path:"/accounts/${goodId2}")
+
+        then:
+        response.status == 200
+        response.data
+        response.data.followerCount == 0
+        response.data.followingCount == 0
+        response.data.id == goodId2
+    }
+
+    def '#goodId1 can follow #goodId2'() {
+        when:
+        def response = restClient.get(path:"/accounts/${goodId1}")
+
+        then:
+        response.status == 200
+        response.data
+        response.data.followerCount == 0
+        response.data.followingCount == 0
+
+        when:
+        response = restClient.post(path:"/accounts/${goodId1}/startFollowing", query: [followAccount: goodId2],
+                contentType: 'application/json', body:[])
+
+        then:
+        response.status == 201
+        response.data
+        response.data.id == goodId1
+        response.data.followerCount == 0
+        response.data.followingCount == 1
+
+        when:
+        response = restClient.get(path:"/accounts/${goodId1}")
+
+        then:
+        response.status == 200
+        response.data
+        response.data.id == goodId1
+        response.data.followerCount == 0
+        response.data.followingCount == 1
+
+        when:
+        response = restClient.get(path:"/accounts/${goodId1}/following")
+
+        then:
+        response.status == 200
+        response.data
+        response.data.size() == 1
+
+        when:
+        def data = response.data.find { it -> it.id == goodId2 }
+
+        then:
+        data
+        data.followingCount == 0
+        data.followerCount == 1
+
+        when:
+        response = restClient.get(path:"/accounts/${goodId2}")
+
+        then:
+        response.status == 200
+        response.data
+        response.data.id == goodId2
+        response.data.followerCount == 1
+        response.data.followingCount == 0
+
+        when:
+        response = restClient.get(path:"/accounts/${goodId2}/followers")
+
+        then:
+        response.status == 200
+        response.data
+        response.data.size() == 1
+
+        when:
+        data = response.data.find { it -> it.id == goodId1 }
```

Comment:
Mostly because I'll use that same pattern when there are more than one item returned.  In this case, it doesn't matter, but if there were two followers returned, then this is how they would be searched.